homeassistant/loader.py

```diff
@@ -63,6 +63,77 @@ def manifest_from_legacy_module(domain: str, module: ModuleType) -> Dict:
     }
 
 
+def manifest_from_path(path: pathlib.Path) -> Dict:
+    """Try to load a manifest from a component path.
+
+    Will return None if no manifest exist or if it is malformed
+    """
+    try:
+        manifest_path = path / "manifest.json"
+        with manifest_path.open() as data:
+            manifest = json.load(data)
+            return manifest
```

Comment:
I was thinking of doing a schema validation here to avoid surprises

Reply:
Maybe inside the resolve from root method but only if not built in

Reply:
Actually let's not do it. We expect code to be correct format. We should make hassfest check custom components too instead